### PR TITLE
fix: use tseslint v8(rc) + eslint v9

### LINKED ISSUE #120: Bug: ESLint config cli fails - eslint@9.3.0 vs 8.56.0
### Environment

Node version: v20.13.1
npm version: 10.5.2
Local ESLint version: 
Global ESLint version: 
Operating System: mcr.microsoft.com/devcontainers/typescript-node:1-20-bookworm

### What parser are you using?

Other

### What did you do?

<details>
<summary>Configuration</summary>

```
npm init @eslint/config@latest -- --debug
```
</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

Create config file and install dependencies.

### What actually happened?

✔ How would you like to use ESLint? · problems
✔ What type of modules does your project use? · commonjs
✔ Which framework does your project use? · none
✔ Does your project use TypeScript? · typescript
✔ Where does your code run? · browser
The config that you've selected requires the following dependencies:

eslint@9.x, globals, @eslint/js, typescript-eslint
✔ Would you like to install them now? · No / Yes
✔ Which package manager do you want to use? · npm
☕️Installing...
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ts-common@1.0.0
npm ERR! Found: eslint@9.3.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"9.x" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^8.56.0" from typescript-eslint@7.10.0
npm ERR! node_modules/typescript-eslint
npm ERR!   dev typescript-eslint@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/node/.npm/_logs/2024-05-21T20_57_05_935Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /home/node/.npm/_logs/2024-05-21T20_57_05_935Z-debug-0.log
A config file was generated, but the config file itself may not follow your linting rules.

### Link to Minimal Reproducible Example

https://github.com/diberry/eslint9-examples/tree/diberry/feature-examples/ts-common

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

The config file was created.

### PR DESCRIPTION
**Pull request by aladdin-add**
it's another way to fix https://github.com/eslint/create-config/issues/89#issuecomment-2118575800

fixes #120

### CODE REVIEW

File: lib/config-generator.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: plug
         }
 
         if (this.answers.language === "typescript") {
-            this.result.devDependencies.push("typescript-eslint");
+            this.result.devDependencies.push("typescript-eslint@rc-v8");
```
Comment (mdjermanovic) [inline lib/config-generator.js:169]: I'm not sure if we should be installing prerelease versions. Maybe rather wait a bit for the final v8.0.0?

Comment (aladdin-add) [reply lib/config-generator.js:169]: the problem: eslint v9 + tseslint(latest) could cause an installing error.

I also prepared a fix to use `npm install --force` to fix it: https://github.com/eslint/create-config/pull/118. Which one do you prefer?

Comment (aladdin-add) [reply lib/config-generator.js:169]: /cc @nzakas @fasttime

Comment (nzakas) [reply lib/config-generator.js:169]: I'd use `--force` for now.

**Pull request closed without merging**